quickwit/quickwit-actors/src/spawn_builder.rs

```diff
@@ -284,8 +284,9 @@ impl<A: Actor> ActorExecutionEnv<A> {
             while let Some(envelope) = try_recv_envelope(&mut self.inbox, &self.ctx) {
                 self.process_one_message(envelope).await?;
             }
-            self.ctx.yield_now().await;
-
+            if self.actor.get_mut().yield_after_each_message() {
```

Comment:
What is the motivation here? It seems like a good idea to yield here.

Reply:
This fixes the issue that Indexer and DocProcessor run on one thread

Reply:
That's not a good sign :-/